### ISSUE TITLE
Cannot use static addresses using adapters with support for dual mode

### ISSUE THREAD

**Issue by SebastianForslund** [bluez/bluez#756]
I would like my Bluetooth device to use static addresses. After setting the ControllerMode to "le", the public address to 00:00:00:00:00:00, and creating an appropriate "/var/lib/bluetooth/addresses" file, Bluez still refuses to set the static address. 

When the static address is about to be set, a check on the **supported** Bluetooth modes is made: 
https://github.com/bluez/bluez/blob/master/src/adapter.c#L9846

Since I have a controller with dual mode support, this check fails. Would it be possible to check the current configuration through the btd_opts instead? That way dual mode adapters can still use static addresses by setting the ControllerMode to "le" in main.conf.

**Comment by Vudentz** [comment]
@SebastianForslund problem is that if we do allow that then it can break pairing when you switch back BR/EDR which is why that check exists in the first place, the proper way to enable usage of random addresses with dual-mode controllers is to have privacy enabled.

**Comment by SebastianForslund** [comment]
Thanks for the quick answer.

I was hoping I could use no public address with just a random static address that i generate. 

From what I read in this file, it seems that a dual-mode adapters must always have an underlying public address: https://github.com/bluez/bluez/blob/master/doc/mgmt-api.txt#L307

I suppose the only way to do this is if there is a vendor specific command to disable the dual-mode completely. Or is there another way? If the answer to that question is no, I suppose we can close this issue.

Thanks for your time :)

**Comment by Vudentz** [comment]
@SebastianForslund have you tried setting Privacy=yes? That should take care of using RPAs when advertising, etc.

**Comment by SebastianForslund** [comment]
Yes, the RPA:s work fine, but it's not why I want to use a static random address. The reason i'm trying to set it myself is that I want control over which MAC address I use in the pairing process. This could of course be done with a public address but I want to avoid using those.
